components/user/UserEvents.js

```diff
@@ -1,38 +1,92 @@
 import { useState } from "react";
-
 import EventCard from "@components/event/EventCard";
 import Alert from "@components/Alert";
-import EventKey from "@components/event/EventKey";
+import DropdownMenu from "@components/form/DropDown";
 
 export default function UserEvents({ data }) {
-  const [eventType, setEventType] = useState("future");
-  const futureEvents = data.events.filter((event) => event.date.future);
-
-  let categorizedEvents = {
-    future: futureEvents,
-    ongoing: data.events.filter((event) => event.date.ongoing === true),
-    virtual: futureEvents.filter((event) => event.isVirtual === true),
-    inPerson: futureEvents.filter((event) => event.isInPerson === true),
-    cfpOpen: futureEvents.filter((event) => event.date.cfpOpen === true),
-    free: data.events.filter((event) => event.price?.startingFrom === 0),
-    paid: data.events.filter((event) => event.price?.startingFrom > 0),
-    past: data.events.filter((event) => event.date.future === false),
+  const [eventType, setEventType] = useState("all");
+
+  const eventOptions = [
+    { value: 'all' , name: 'All Events'},
+    { value: 'future' , name: 'Future Events'},
+    { value: 'ongoing', name: 'Ongoing Events'},
+    { value: 'virtual', name: 'Virtual Events'},
+    { value: 'inPerson', name: 'In-Person Events'},
+    { value: 'cfpOpen', name:'Events with open CFP'},
+    { value: 'free', name: 'Free Events'},
+    { value: 'paid', name: 'Paid Events'},
+    { value: 'past', name: 'Past Events'}
+  ];
+
+  const handleEventTypeChange = (event) => {
+    setEventType(event.target.value);
+  };
+
+  const filterByEventType = (event, eventType) => {
```

Comment:
Doing Date math inside a server-side rendered component is a bad idea.

Please change this section to look like:
```js
  const filterByEventType = (event, eventType) => {
    switch (eventType) {
      case "future":
        return event.date.future;
      case "ongoing":
        return event.date.ongoing;
      case "virtual":
        return event.date.future && event.isVirtual;
      case "inPerson":
        return event.date.future && event.isInPerson;
      case "cfpOpen":
        return event.date.cfpOpen;
      case "free":
        return event.price?.startingFrom === 0;
      case "paid":
        return event.price?.startingFrom > 0;
      case "past":
        return !event.date.future;
      default:
        return true;
    }
  };
```

Reply:
@dan-mba I have changed the function to lool like what you suggested